substrafl_examples/go_further/run_diabetes_substrafl.py

```diff
@@ -0,0 +1,562 @@
+                    _algo_name=f"Computing second order analytics with {self.__class__.__name__}",
+                ),
+                local_state=local_states[node.organization_id],
+                round_idx=1,
+                authorized_ids=set([node.organization_id]),
+                aggregation_id=aggregation_node.organization_id,
+                clean_models=False,
+            )
+
+            # All local analytics are stored in the second_order_shared_states,
+            # given as input the the aggregation method.
+            second_order_shared_states.append(next_shared_state)
+
+        # Call the aggregation method on the second_order_shared_states
+        self.second_order_aggregated_state = aggregation_node.update_states(
+            self.aggregation(
+                shared_states=second_order_shared_states,
+                _algo_name="Aggregating second order",
+            ),
+            round_idx=1,
+            authorized_ids=set([train_data_node.organization_id for train_data_node in train_data_nodes]),
+            clean_models=False,
+        )
+
+    def save_local_state(self, path: pathlib.Path):
+        """This function will save the important local state to retrieve after each new
+        call to a train or test task.
+
+        Args:
+            path (pathlib.Path): Path where to save the local_state. Provided internally by
+                Substra.
+        """
+        state_to_save = {
+            "first_order": self.first_order_aggregated_state,
+            "second_order": self.second_order_aggregated_state,
+        }
+        with open(path, "w") as f:
+            json.dump(state_to_save, f)
+
+    def load_local_state(self, path: pathlib.Path):
+        """Mirror function to load the local_state from a file saved using
+        ``save_local_state``.
+
+        Args:
+            path (pathlib.Path): Path where to load the local_state. Provided internally by
+                Substra.
+
+        Returns:
+            ComputePlanBuilder: return self with the updated local state.
+        """
+        with open(path, "r") as f:
+            state_to_load = json.load(f)
+
+        self.first_order_aggregated_state = state_to_load["first_order"]
+        self.second_order_aggregated_state = state_to_load["second_order"]
+
+        return self
+
+
+# %%
+# Now that we saw the implementation of the custom ``Analytics`` class, we can add detailed to some of the previously
+# introduced concept.
+#
+# The ``update_state`` method outputs the new state of the node, that can be passed as an argument to a following one.
+# This succession of ``next_state`` pass to new ``node.update_state`` is how Substra build the graph of the
+# compute plan.
+#
+# The ``load_local_state`` and ``save_local_state`` are two methods used at each new iteration on a Node, in order to
+# retrieve a the previous local state that have not been shared with the other ``Nodes``.
```

Comment:
a / the ⚔️

Reply:
Sorry I don't get the comment 😶

Reply:
sorry I wasn't very clear. It's written `retrieve a the previous local state`, and I think one of the two articles is enough :)

Reply:
Aaaaaah ahah :D thanks !!